src/main/java/net/javadiscord/javabot/systems/help/AutoCodeFormatter.java

```diff
@@ -0,0 +1,164 @@
+
+/**
+ * Handles auto-formatting of code in help-channels.
+ * A non-listener class since it has to be called strictly after {@link HelpListener} to not send its messages before channel reserving.
+ */
+@RequiredArgsConstructor
+@Component
+public class AutoCodeFormatter {
+	private final AutoMod autoMod;
+	private final BotConfig botConfig;
+	private final UserPreferenceService preferenceService;
+
+	/**
+	 * Method responsible for finding a place to insert a codeblock, if present.
+	 *
+	 * @param event a {@link MessageReceivedEvent}.
+	 * @return a MessageCodeblock instance, holding a startIndex, content and
+	 * an endIndex. Returns null if no place was found.
+	 */
+	@Nullable
+	private static CodeBlock findCodeblock(@NotNull MessageReceivedEvent event) {
+		String msg = event.getMessage().getContentRaw();
+		int openingBracket = msg.indexOf("{");
+		int closingBracket = msg.lastIndexOf("}");
+		if (closingBracket == -1 || openingBracket == -1) {
+			return null;
+		}
+		int startIndex = msg.lastIndexOf("\n", openingBracket);
+		int endIndex = msg.indexOf("\n", closingBracket);
+		if (startIndex == -1) {
+			startIndex = 0;
+		}
+		if (endIndex == -1) {
+			endIndex = msg.length();
+		}
+		return new CodeBlock(startIndex, endIndex);
+	}
+
+	/**
+	 * called by {@link HelpListener#onMessageReceived(MessageReceivedEvent)} on every message.
+	 * It is worth noting that this class can't register as an event handler itself due to there being no way of
+	 * setting its methods to be strictly called after {@link HelpListener}.
+	 *
+	 * @param event          a {@link MessageReceivedEvent}
+	 * @param isFirstMessage flag that should be set if the message is a thread-opening one.
+	 */
+	void handleMessageEvent(@Nonnull MessageReceivedEvent event, boolean isFirstMessage) {
+		if (!event.isFromGuild()) {
+			return;
+		}
+		if (event.getAuthor().isBot() || event.getMessage()
+				.getAuthor()
+				.isSystem()) {
+			return;
+		}
+		if (autoMod.hasSuspiciousLink(event.getMessage()) ||
+				autoMod.hasAdvertisingLink(event.getMessage())) {
+			return;
+		}
+		if (event.isWebhookMessage()) {
+			return;
+		}
+		if (!event.isFromThread()) {
+			return;
+		}
+		if (event.getChannel()
+				.asThreadChannel()
+				.getParentChannel()
+				.getIdLong() != botConfig.get(event.getGuild())
+				.getHelpConfig()
+				.getHelpForumChannelId()) {
+			return;
+		}
+		if (!Boolean.parseBoolean(preferenceService.getOrCreate(Objects.requireNonNull(event.getMember())
+				.getIdLong(), Preference.FORMAT_UNFORMATTED_CODE).getState())) {
+			return;
+		}
+
+
+		if (event.getMessage().getContentRaw().contains("```")) {
+			return; // exit if already contains codeblock
+		}
+
+		CodeBlock code = findCodeblock(event);
+		if (code == null) {
+			return;
+		}
+
+		if (isFirstMessage || !event.getMessage().getMentions().getUsers().isEmpty() ||
+				!event.getMessage().getMentions().getRoles().isEmpty() ||
+				!event.getMessage().getMentions().mentionsEveryone()) {
```

Comment:
This negation is false. It should send the format hint if `@everyone` is mentioned.